### PR TITLE
Fix of props update order, inactive while updating if data changed 

### PR DESCRIPTION
**Pull request by iSuslov**
#148  
#185 
#148 
#164
Please see https://github.com/PolymerElements/app-route/issues/164#issuecomment-366106287

### CODE REVIEW

**Comment by iSuslov** [comment]
I signed it!

**Comment by e111077** [comment]
Thanks for the submission. Unfortunately this is breaking some of our integration and unit tests.

I think the right approach would be to unfortunately do another hack like `__setMulti`. Perhaps running the [internals of `setProperites`](https://github.com/Polymer/polymer/blob/4c8d25551e200733c73aff297354b4080c1d1427/lib/mixins/property-effects.html#L1629) in the correct order may notify in the desired order.

i.e. run `_setPendingPropertyOrPath` for each of the new values in the correct order and call `_invalidateProperties`. If that doesn't work we may have to dig in more and find the correct internal notification functions.

**Comment by iSuslov** [comment]
Hi, your idea is reasonable and I will try it. I made another commit, which probably will fail unit testing again, the idea is to set `app-route` to inactive state while updating properties **if** `data` changed and then bring it back to `active`. Why it is important: see updated https://github.com/PolymerElements/app-route/issues/164#issuecomment-366106287

**Comment by iSuslov** [comment]
Well, using internals of 'setProperties' does not help really, since it is still unordered on lower level https://github.com/Polymer/polymer/blob/5c02730974f8e3138405e596cb5f8d8bb388a41f/lib/mixins/property-effects.html#L124

File: app-route.html [hunk 1]
```diff
@@ -318,8 +317,11 @@
         }
 
         if (this.setProperties) {
-          // atomic update
-          this.setProperties(propertyUpdates, true);
+            //Update properties in a specific order: data, tail and set inactive during update if data changed
+            propertyUpdates.hasOwnProperty("data") && this._setPendingPropertyOrPath("data", propertyUpdates.data, true);
```
Comment (e111077) [inline app-route.html]: What about setting without notifying all of the properties:

`this._setPendingPropertyOrPath("data", propertyUpdates.data, false);`

and then call `this.notifyPath('data', propertyUpdates.data)` but in the correct order after all have been set?

Comment (e111077) [reply app-route.html]: Not 100% sure this will work; not sure if it will trigger `runEffects`

**Comment by e111077** [comment]
Any updates?

**Comment by mercmobily** [comment]
Ho there. I am following this issue from #164 which is still basically unresolved... and it looks like it will be for a while. Any hints on this?

**Pull request closed without merging**